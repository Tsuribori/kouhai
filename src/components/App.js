import React, { Component } from 'react';
import PropTypes from 'prop-types';
import scroll from 'react-scroll';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import { AppBar, Toolbar, IconButton, Menu, MenuItem, CssBaseline, Fab } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import { ArrowUpward } from '@material-ui/icons';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';
import PopularList from './PopularList.js';
import Detail from './Detail.js';
import Schedule from './Schedule.js';
import Top from './Top.js';
import Search from './Search.js';

const styles = theme => ({
  fab: {
    position: "fixed",
    bottom: theme.spacing.unit * 2,
    right: theme.spacing.unit * 2,
  },
  search: {
    display: "flex",
    position: "relative",
    width: "100%",
    alignItems: "center",
    justifyContent: "flex-end", 
  },
});

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      anchorEl: null,
      searchPage: false,
    }
    this.handleMenu = this.handleMenu.bind(this); 
  }

  handleMenu(event) {
    this.setState({ anchorEl: event.currentTarget });
  }

  handleClose = () => {
    this.setState({ anchorEl: null });
  }

  handleClick = () => {
    scroll.animateScroll.scrollToTop();
  }

  handleIconHide = () => {
    this.setState({ searchPage: !this.state.searchPage });
  }

  render() {
    const { anchorEl } = this.state;
    const { classes } = this.props;
    const open = Boolean(anchorEl);
    const paths = [
      {
        url: '/',
        name: 'Popular'
      },
      {
        url: '/top',
        name: 'Top'
      },
      {
       url: '/schedule',
       name: 'Schedule'
      }
    ];
    return (
      <Router>
        <CssBaseline>
          <div>
            <AppBar position="static">
              <Toolbar>
                <IconButton
                  color="inherit"
                  onClick={this.handleMenu}
                >
                  <MenuIcon />
                </IconButton>
                <Menu
                  anchorEl={anchorEl}
                  open={open}
                  onClose={this.handleClose}
                >
                  {paths.map((path) => (
                    <MenuItem 
                      key={path.url}
                      component={Link} 
                      to={path.url}
                      onClick={this.handleClose}
                    >
                      {path.name}
                    </MenuItem>
                  ))};
                </Menu>
                {!this.state.searchPage &&
                <div className={classes.search}>
                  <IconButton
                    component={Link}
                    to="/search"
                    color="inherit"
                  >
                    <SearchIcon />
                  </IconButton>
                </div>
                }
              </Toolbar>
            </AppBar>
          </div>
          <div>
            <Route exact path="/" component={PopularList} />
            <Route path="/anime/:mal_id(\d*)" component={Detail} />
            <Route path="/top" component={Top} />
            <Route exact path="/schedule" component={Schedule} />
            <Route exact path="/search" 
              render={(props) => <Search {...props} handleHide={this.handleIconHide}/>}
            />
          </div>
          <Fab onClick={this.handleClick} className={classes.fab} color="primary">
            <ArrowUpward />
          </Fab>
        </CssBaseline>
      </Router>
    );
  }
}

App.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(App);
