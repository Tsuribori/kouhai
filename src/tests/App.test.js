import React from 'react';
import ReactDOM from 'react-dom';
import { Menu } from '@material-ui/icons';
import App from '../components/App';
import './TestSetUp.js';
import { shallow }Â from 'enzyme';
import toJson from 'enzyme-to-json';

describe('App test', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('Test App snapshot', () => {
    const tree = shallow(<App />).dive();
    expect(toJson(tree)).toMatchSnapshot();
    const event = {currentTarget: <Menu />};
    tree.instance().handleMenu(event);
    expect(toJson(tree)).toMatchSnapshot();
    tree.instance().handleClose();
    expect(toJson(tree)).toMatchSnapshot();
  });
});
