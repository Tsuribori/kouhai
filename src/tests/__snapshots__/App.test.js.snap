// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test Test App snapshot 1`] = `
<BrowserRouter>
  <WithStyles(CssBaseline)>
    <div>
      <WithStyles(AppBar)
        position="static"
      >
        <WithStyles(Toolbar)>
          <WithStyles(IconButton)
            color="inherit"
            onClick={[Function]}
          >
            <pure(MenuIcon) />
          </WithStyles(IconButton)>
          <WithStyles(Menu)
            anchorEl={null}
            onClose={[Function]}
            open={false}
          >
            <WithStyles(MenuItem)
              component={[Function]}
              key="/"
              onClick={[Function]}
              to="/"
            >
              Popular
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              component={[Function]}
              key="/top"
              onClick={[Function]}
              to="/top"
            >
              Top
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              component={[Function]}
              key="/schedule"
              onClick={[Function]}
              to="/schedule"
            >
              Schedule
            </WithStyles(MenuItem)>
            ;
          </WithStyles(Menu)>
          <div
            className="App-search-192"
          >
            <WithStyles(IconButton)
              color="inherit"
              component={[Function]}
              to="/search"
            >
              <pure(SearchIcon) />
            </WithStyles(IconButton)>
          </div>
        </WithStyles(Toolbar)>
      </WithStyles(AppBar)>
    </div>
    <div>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/anime/:mal_id(\\\\d*)"
      />
      <Route
        component={[Function]}
        path="/top"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/schedule"
      />
      <Route
        exact={true}
        path="/search"
        render={[Function]}
      />
    </div>
    <WithStyles(Fab)
      className="App-fab-191"
      color="primary"
      onClick={[Function]}
    >
      <pure(ArrowUpwardIcon) />
    </WithStyles(Fab)>
  </WithStyles(CssBaseline)>
</BrowserRouter>
`;

exports[`App test Test App snapshot 2`] = `
<BrowserRouter>
  <WithStyles(CssBaseline)>
    <div>
      <WithStyles(AppBar)
        position="static"
      >
        <WithStyles(Toolbar)>
          <WithStyles(IconButton)
            color="inherit"
            onClick={[Function]}
          >
            <pure(MenuIcon) />
          </WithStyles(IconButton)>
          <WithStyles(Menu)
            anchorEl={<pure(MenuIcon) />}
            onClose={[Function]}
            open={true}
          >
            <WithStyles(MenuItem)
              component={[Function]}
              key="/"
              onClick={[Function]}
              to="/"
            >
              Popular
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              component={[Function]}
              key="/top"
              onClick={[Function]}
              to="/top"
            >
              Top
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              component={[Function]}
              key="/schedule"
              onClick={[Function]}
              to="/schedule"
            >
              Schedule
            </WithStyles(MenuItem)>
            ;
          </WithStyles(Menu)>
          <div
            className="App-search-192"
          >
            <WithStyles(IconButton)
              color="inherit"
              component={[Function]}
              to="/search"
            >
              <pure(SearchIcon) />
            </WithStyles(IconButton)>
          </div>
        </WithStyles(Toolbar)>
      </WithStyles(AppBar)>
    </div>
    <div>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/anime/:mal_id(\\\\d*)"
      />
      <Route
        component={[Function]}
        path="/top"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/schedule"
      />
      <Route
        exact={true}
        path="/search"
        render={[Function]}
      />
    </div>
    <WithStyles(Fab)
      className="App-fab-191"
      color="primary"
      onClick={[Function]}
    >
      <pure(ArrowUpwardIcon) />
    </WithStyles(Fab)>
  </WithStyles(CssBaseline)>
</BrowserRouter>
`;

exports[`App test Test App snapshot 3`] = `
<BrowserRouter>
  <WithStyles(CssBaseline)>
    <div>
      <WithStyles(AppBar)
        position="static"
      >
        <WithStyles(Toolbar)>
          <WithStyles(IconButton)
            color="inherit"
            onClick={[Function]}
          >
            <pure(MenuIcon) />
          </WithStyles(IconButton)>
          <WithStyles(Menu)
            anchorEl={null}
            onClose={[Function]}
            open={false}
          >
            <WithStyles(MenuItem)
              component={[Function]}
              key="/"
              onClick={[Function]}
              to="/"
            >
              Popular
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              component={[Function]}
              key="/top"
              onClick={[Function]}
              to="/top"
            >
              Top
            </WithStyles(MenuItem)>
            <WithStyles(MenuItem)
              component={[Function]}
              key="/schedule"
              onClick={[Function]}
              to="/schedule"
            >
              Schedule
            </WithStyles(MenuItem)>
            ;
          </WithStyles(Menu)>
          <div
            className="App-search-192"
          >
            <WithStyles(IconButton)
              color="inherit"
              component={[Function]}
              to="/search"
            >
              <pure(SearchIcon) />
            </WithStyles(IconButton)>
          </div>
        </WithStyles(Toolbar)>
      </WithStyles(AppBar)>
    </div>
    <div>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/anime/:mal_id(\\\\d*)"
      />
      <Route
        component={[Function]}
        path="/top"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/schedule"
      />
      <Route
        exact={true}
        path="/search"
        render={[Function]}
      />
    </div>
    <WithStyles(Fab)
      className="App-fab-191"
      color="primary"
      onClick={[Function]}
    >
      <pure(ArrowUpwardIcon) />
    </WithStyles(Fab)>
  </WithStyles(CssBaseline)>
</BrowserRouter>
`;
